res/controllers/Numark-Mixtrack-Platinum-scripts.js

```diff
@@ -0,0 +1,1415 @@
+                var effectMatch = component.group.match(script.individualEffectRegEx);
+                if (effectMatch !== null) {
+                    component.group = '[EffectRack1_EffectUnit' +
+                                      eu.currentUnitNumber +
+                                      '_Effect' + effectMatch[2] + ']';
+                }
+            }
+        });
+    };
+
+    this.setCurrentUnit(unitNumbers[0]);
+
+    this.dryWetKnob = new components.Encoder({
+        group: this.group,
+        inKey: 'mix',
+        input: function (channel, control, value, status, group) {
+            if (value === 1) {
+                this.inSetParameter(this.inGetParameter() + 0.05);
+            } else if (value === 127) {
+                this.inSetParameter(this.inGetParameter() - 0.05);
+            }
+        },
+    });
+
+    this.EffectUnitTouchStrip = function() {
+        components.Pot.call(this);
+        this.firstValueRecived = true;
+        this.connect();
+    };
+    this.EffectUnitTouchStrip.prototype = new components.Pot({
+        relative: true, // this disables soft takeover
+        input: function (channel, control, value, status, group) {
+            // never do soft takeover when the touchstrip is used
+            engine.softTakeover(this.group, this.inKey, false);
+            components.Pot.prototype.input.call(this, channel, control, value, status, group);
+        },
+        connect: function() {
+            this.focus_connection = engine.makeConnection(eu.group, "focused_effect", this.onFocusChange);
+            this.focus_connection.trigger();
+        },
+        disconnect: function() {
+            this.focus_connection.disconnect();
+        },
+        onFocusChange: function(value, group, control) {
+            if (value === 0) {
+                this.group = eu.group;
+                this.inKey = 'super1';
+            }
+            else {
+                this.group = '[EffectRack1_EffectUnit' + eu.currentUnitNumber + '_Effect' + value + ']';
+                this.inKey = 'meta';
+            }
+        },
+    });
+
+    this.BpmTapButton = function () {
+        this.group = '[Channel' + eu.currentUnitNumber + ']';
+        this.midi = [0x97 + eu.currentUnitNumber, 0x04];
+        components.Button.call(this);
+    };
+    this.BpmTapButton.prototype = new components.Button({
+        type: components.Button.prototype.types.push,
+        key: 'bpm_tap',
+        off: 0x01,
+        connect: function () {
+            this.group = '[Channel' + eu.currentUnitNumber + ']';
+            components.Button.prototype.connect.call(this);
+        },
+        input: function (channel, control, value, status, group) {
+            components.Button.prototype.input.call(this, channel, control, value, status, group);
+            if (this.isPress(channel, control, value, status)) {
+                eu.forEachComponent(function (component) {
+                    if (component.tap !== undefined) {
```

Comment:
Not strictly necessary, but a good safety measure
```suggestion
                    if (component.tap !== undefined && typeof component.tap === 'function') {
```